### PR TITLE
Merge vs16.2 into master

### PR DESCRIPTION
**Pull request by livarcocc**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -15,7 +15,8 @@
     <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
   </PropertyGroup>
   <PropertyGroup>
-    <VersionPrefix>16.3.0</VersionPrefix>
+    <MSBuildVersionMajorMinor>16.3</MSBuildVersionMajorMinor>
+    <VersionPrefix>$(MSBuildVersionMajorMinor).0</VersionPrefix>
```
Comment (rainersigwald) [inline eng/Versions.props]: Can you revert this block? It's no longer needed because we can hack less below.

File: eng/Versions.props [hunk 2]
```diff
@@ -37,4 +38,35 @@
...
+         In the future we'll want this to start with $(VersionPrefix), but
+         since 16.2 shipped with a high version, we need to extract a new
+         higher version (generated by Arcade as the fourth part of the
+         version).
+
+         So: we want this to match the NuGet package version and also the
+         AssemblyInformationalVersion. Jump through hoops to do so.
+         -->
+    <PropertyGroup>
+      <FileVersion>$(MSBuildVersionMajorMinor).$(FileVersion.Split('.')[3]).0</FileVersion>
```
Comment (rainersigwald) [inline eng/Versions.props]:
```suggestion
      <FileVersion>$(VersionPrefix).$(FileVersion.Split('.')[3])</FileVersion>
```

Comment (rainersigwald) [reply eng/Versions.props]: I think this is a better format long-term.

Comment (livarcocc) [reply eng/Versions.props]: But the version prefix has all 3 parts. You are ok with that?

Comment (rainersigwald) [reply eng/Versions.props]: Yes, I think it makes the most sense to have FileVersion start with `Major.Minor.Revision` and end with the disambiguator.

Comment (rainersigwald) [reply eng/Versions.props]: (But I'm open to hearing other thoughts!)

**Comment by livarcocc** [comment]
I tried this locally, MSBuildVersion is coming out as 16.3.0.

**Pull request merged**